frontend/src/metabase/alert/alert.js

```diff
@@ -0,0 +1,147 @@
+    return async (dispatch, getState) => {
+        dispatch.action(FETCH_ALERTS_FOR_QUESTION_CLEAR_OLD_ALERTS, questionId)
+        await dispatch(fetchAlertsForQuestionRequest.trigger({ questionId }))
+        dispatch.action(FETCH_ALERTS_FOR_QUESTION)
+    }
+}
+
+export const CREATE_ALERT = 'metabase/alerts/CREATE_ALERT'
+const createAlertRequest = new RestfulRequest({
+    endpoint: AlertApi.create,
+    actionPrefix: CREATE_ALERT,
+    storeAsDictionary: true
+})
+export const createAlert = (alert) => {
+    return async (dispatch, getState) => {
+        // TODO: How to handle a failed creation and display it to a user?
+        // Maybe RestfulRequest.trigger should throw an exception
+        // that the React component calling createAlert could catch ...?
+        await dispatch(createAlertRequest.trigger(alert))
+
+        dispatch(addUndo(createUndo({
+            type: "create-alert",
+            // eslint-disable-next-line react/display-name
+            message: () => <div className="flex align-center text-bold"><Icon name="alertConfirm" size="19" className="mr2 text-success" />Your alert is all set up.</div>,
+            action: null // alert creation is not undoable
+        })));
+
+        dispatch.action(CREATE_ALERT)
+    }
+}
+
+export const UPDATE_ALERT = 'metabase/alerts/UPDATE_ALERT'
+const updateAlertRequest = new RestfulRequest({
+    endpoint: AlertApi.update,
+    actionPrefix: UPDATE_ALERT,
+    storeAsDictionary: true
+})
+export const updateAlert = (alert) => {
+    return async (dispatch, getState) => {
+        await dispatch(updateAlertRequest.trigger(alert))
+
+        dispatch(addUndo(createUndo({
+            type: "update-alert",
+            // eslint-disable-next-line react/display-name
+            message: () => <div className="flex align-center text-bold"><Icon name="alertConfirm" size="19" className="mr2 text-success" />Your alert was updated.</div>,
+            action: null // alert updating is not undoable
+        })));
+
+        dispatch.action(UPDATE_ALERT)
+    }
+}
+
+export const UNSUBSCRIBE_FROM_ALERT = 'metabase/alerts/UNSUBSCRIBE_FROM_ALERT'
+export const UNSUBSCRIBE_FROM_ALERT_CLEANUP = 'metabase/alerts/UNSUBSCRIBE_FROM_ALERT_CLEANUP'
+const unsubscribeFromAlertRequest = new RestfulRequest({
+    endpoint: AlertApi.unsubscribe,
+    actionPrefix: UNSUBSCRIBE_FROM_ALERT,
+    storeAsDictionary: true
+})
+export const unsubscribeFromAlert = (alert) => {
+    return async (dispatch, getState) => {
+        await dispatch(unsubscribeFromAlertRequest.trigger(alert))
+        dispatch.action(UNSUBSCRIBE_FROM_ALERT)
+
+        // This delay lets us to show "You're unsubscribed" text in place of an
+        // alert list item for a while before removing the list item completely
+        setTimeout(() => dispatch.action(UNSUBSCRIBE_FROM_ALERT_CLEANUP, alert.id), 5000)
+    }
+}
+
+export const DELETE_ALERT = 'metabase/alerts/DELETE_ALERT'
+const deleteAlertRequest = new RestfulRequest({
```

Comment:
what is a RestfulRequest? 😱 

I thought we were supposed to be doing like thunk actions and promises and w/e